### DISCUSSION TITLE
Sunbird Cokreat - (Contribution Service )

### DISCUSSION THREAD

**Discussion by agamdahiya** [Sunbird-coKreat/Community#73] [Q&A]
I was facing the issue in the setup Contribution service of the Sunbird Cokreat Building Block in the Kafka part . 
Provide me the suitable information regarding this. Find the screenshot of the error for a better understanding.

![image](https://user-images.githubusercontent.com/127106035/231179588-18028bc7-4996-4586-bc9a-13aefba315f5.png)

**Comment by mmanimegalai** [comment]
Tagging @arun1205 @vaivk369 cc @vinukumar-vs

**Comment by arun1205** [comment]
@agamdahiya looking into it and will let u know

**Reply by arun1205** [reply]
@agamdahiya seems u r running master branch.. try switching to latest branch 5.2.0 or later. Let me know if u still face the error

**Reply by agamdahiya** [reply]
@arun1205 I am getting same error even switching to latest branch 5.2.1 . 
![image](https://user-images.githubusercontent.com/127106035/231382953-6a6561ae-21d4-4045-b5e3-1df01a1aab1f.png)

**Reply by arun1205** [reply]
@vaivk369 as discussed can u check and update @agamdahiya 
@agamdahiya can u share the env file u r using?

**Reply by agamdahiya** [reply]
[Env variables.txt](https://github.com/Sunbird-coKreat/Community/files/11213627/Env.variables.txt)

This is env file i was using .

**Reply by vaivk369** [reply]
@agamdahiya i m checking this at my end. Will get back to you on this tomorrow.

**Reply by vaivk369** [reply]
@agamdahiya can you  please try editing the envVariables.js file and have the default value for **SUNBIRD_AUTO_CREATION_TOPIC** variable?

`    SUNBIRD_AUTO_CREATION_TOPIC: process.env.sunbird_auto_creation_topic || 'dock-test-topic',
`

**Reply by agamdahiya** [reply]
@vaivk369 i had made the changes in the envVariables.js file but still facing the same issue.
Find the screenshot below regarding this issue.

![image](https://user-images.githubusercontent.com/127106035/231660187-cac97b7c-5956-4b3c-a150-62fbe4d33ff6.png)

**Comment by vaivk369** [comment]
Ok. Can you also try to comment out the following line from app.js fiile?
`qumlConsumerService.qumlConsumer();`
Even if you see kafka error, you should be able to create projects through APIs.

**Reply by agamdahiya** [reply]
@vaivk369  i am able to run the application. But the kafka issue remains the same and also had a query whats the purpose of commenting out this -- qumlConsumerService.qumlConsumer(); from app.js file ?
Find the screenshot of the issues that i am facing now .
![image](https://user-images.githubusercontent.com/127106035/231767147-65dbd4bb-a064-4849-9e39-9b1cc3d95df6.png)

**Reply by arun1205** [reply]
@agamdahiya I believe u r using node v 12.x.x as mentioned in the documentation. use 12.20.1 to be precise

**Reply by agamdahiya** [reply]
@arun1205 @vaivk369  Yes i am using node version 12.20.1 it is running completely fine .
 But I had a query regarding middleware token generation . How it is generating ? 
 I required Parameters and Body of the variables for the testing purpose of APIs . Find the screenshot below for those .

![image](https://user-images.githubusercontent.com/127106035/232493826-9fc132a6-6eea-481e-b330-78add4c0a98e.png)

**Reply by arun1205** [reply]
@agamdahiya u can log the incoming request using logger.info at line number 82 which would be something like this

https://documenter.getpostman.com/view/25186239/2s935ivSkP#70d66bec-a8f6-495c-81ce-44724d94554c
